Cannot Bundle Assets on Amazon Linux 2023: Pak Files Don't Contain Manifest.XML

Exporting a project on (Amazon Linux 2023) AL2023 fails due to asset bundler not generating any Manifest.xml

**Steps to reproduce**
1. Install a project on AL2023 
2. Export the project
```
 ~/o3de/scripts/o3de.sh export-project -es ~/multiplayersample/ExportScripts/export_mps.py -out ~/MPSExport_x86 -gl -assets -foa -nogame -noserver -nounified -pl linux -cfg release -ll INFO
```
3. Unzip one of the pak files. Ex: _<export_folder>/Cache/linux/engine_linux.pak_

**Expected behavior**
The pak file should contain a manifest.xml

**Actual behavior**
Manifest.xml doesn't exist

**Hack Workaround**
Generate engine.pak and game.pak on Ubuntu and copy them over to the AL2023 package.

**Found in Branch**
development

https://github.com/o3de/o3de/commit/8d89ec32f88357791283da75424e26197bd6a545
October 3rd 2023


**Desktop/Device (please complete the following information):**
Amazon Linux 2023 x86 machine

Aye, the manifest.xml is failing for sure. Also would be good if it returned an exit code so it was obvious when the export failed.
```
~/multiplayersample/build/tools/bin/profile/AssetBundlerBatch bundles --maxSize 1024 --platform linux --project-path ~/multiplayersample/ --allowOverwrites --outputBundlePath ~/multiplayersample/build/asset_bundling/Bundles/game_linux.pak --assetListFile ~/multiplayersample/build/asset_bundling/AssetLists/game_linux.assetlist

Setting project to ( /home/ec2-user/multiplayersample/ ).
Creating Bundle ( /home/ec2-user/multiplayersample/build/asset_bundling/Bundles/game_linux.pak )...

[ERROR] - ArchiveComponent:
File 'manifest.xml' does not exist inside archive '/home/ec2-user/multiplayersample/build/asset_bundling/Bundles/game_linux.pak'

[ERROR] - AssetBundle:
Failed to extract existing manifest from archive "/home/ec2-user/multiplayersample/build/asset_bundling/Bundles/game_linux.pak".
```

--- a/Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp
+++ b/Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp
@@ -419,14 +419,14 @@ namespace AzToolsFramework
 
             AZ::IO::Path workingPath{ workingDirectory };
             AZ::IO::Path fullPath = workingPath / fileToAdd;
-            AZ::IO::PathView relativePath = AZ::IO::PathView{ fullPath }.LexicallyRelative(workingPath);
+            AZ::IO::Path relativePath = fullPath.LexicallyRelative(workingPath);
 
             AZStd::vector<char> fileBuffer;
             bool success = false;
             if (ArchiveUtils::ReadFile(fullPath, AZ::IO::OpenMode::ModeRead, fileBuffer))
             {
                 int result = archive->UpdateFile(
-                    relativePath.Native(), fileBuffer.data(), fileBuffer.size(), s_compressionMethod,
+                    relativePath.c_str(), fileBuffer.data(), fileBuffer.size(), s_compressionMethod,
                     s_compressionLevel, s_compressionCodec);
 
                 success = (result == AZ::IO::ZipDir::ZD_ERROR_SUCCESS);

